tools/check-napi.js

```diff
@@ -0,0 +1,96 @@
+// Descend into a directory structure and, for each file matching *.node, output
+// based on the imports found in the file whether it's an N-API module or not.
+
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
+
+// Read the output of the command, break it into lines, and use the reducer to
+// decide whether the file is an N-API module or not.
+function checkFile(file, command, arguments, reducer) {
+  const child = child_process.spawn(command, arguments, {
+    stdio: ['inherit', 'pipe', 'inherit']
+  });
+  let leftover = '';
+  let isNapi = undefined;
+  child.stdout.on('data', (chunk) => {
+    if (isNapi === undefined) {
+      chunk = leftover + chunk.toString();
+      const haveLeftover = !!chunk.match(/[\r\n]+$/);
```

Comment:
I think this can be removed as it is not used.

Reply:
Is the chunk **guaranteed** to end on a line boundary?

Reply:
Oh, NM. You meant the variable, not the rest of the logic. You're right.